### ISSUE TITLE
Please consider extending scipy.stats.random_correlation to generate Hermitian matrices

### ISSUE THREAD

**Issue by NeilGirdhar** [scipy/scipy#14325]
I need to generate a random covariance matrix for a multivariate complex normal distribution test.  I used the `scipy.stats.random_correlation` function to generate the covariance in the real case.  Would it be possible to generate random complex Hermitian matrices given a flag to this function?

**Comment by NeilGirdhar** [comment]
Incidentally, does anyone know how to implement this?

**Comment by rkern** [comment]
I don't think it would be appropriate to repurpose that distribution for the complex Hermitian case. Feel free to add one specifically for that case, though.

**Comment by NeilGirdhar** [comment]
Great, I've edited the title.  Unfortunately, I don't know how to implement this.
